app/src/main/java/me/guillaumin/android/osmtracker/db/DataHelper.java

```diff
@@ -65,6 +68,9 @@
 	 */
 	private Context context;
 
+	/** Statistics for all existing tracks */
+	private static TreeMap<Long, TrackStatistics> tracksStatistics = new TreeMap<Long, TrackStatistics> ();
```

Comment:
10k points will take half a second second on 1.3 GHz CPU (it's quad core but I guess that's irrelevant?). In other words, 10 hours worth of recording takes 2 seconds. Which, to be honest, feels 100 times slower than what would be possible with C code and a simple data storage, but that aside – it's good enough for my purposes, and what do you say? Is it worth changing the database?